liblouis/lou_translateString.c

```diff
@@ -82,20 +82,22 @@ typedef struct {
 } OutString;
 
 static int
-putCharacter(widechar c, const TranslationTableHeader *table, int pos, InString input,
-		OutString *output, int *posMapping, int *cursorPosition, int *cursorStatus);
+putCharacter(widechar c, const TranslationTableHeader *table, int pos,
+		const InString *input, OutString *output, int *posMapping, int *cursorPosition,
+		int *cursorStatus);
 static int
-passDoTest(const TranslationTableHeader *table, int pos, InString input, int transOpcode,
-		const TranslationTableRule *transRule, int *passCharDots,
+passDoTest(const TranslationTableHeader *table, int pos, const InString *input,
```

Comment:
Why does the API of the functions passDoTest and passDoAction have a different API. One asks for const InString *input while an other asks for const InString **input

Reply:
passDoAction calls replaceGrouping and removeGrouping, which modify the input buffer.

Reply:
OK, makes sense

Reply:
Previously they modified the input, but now they return a new array.